templates/deno-ts/deno.json

```diff
@@ -0,0 +1,6 @@
+{
+  "compilerOptions": {
+    "lib": ["dom", "dom.iterable", "dom.asynciterable", "deno.ns"]
```

Comment:
Hi @pcattori could you explain why we should customize lib? this line cause the error reported in https://github.com/remix-run/remix/issues/2402

Reply:
I copied it from some of the prior work we had done on Deno, but it might be a vestige from when we were trying to use standard Node tooling + Typescript compiler to produce a Deno bundle. We've since abandoned that approach in favor of writing Deno source code (eventually with Deno tooling like `deno fmt`, `deno lint`, etc... in CI).

Could very well be the case that we should remove the `lib` option.

Reply:
Let's continue the discussion in #2402